pkg/cloud/openstack/machineactuator.go

```diff
@@ -436,18 +441,32 @@ func (oc *OpenstackClient) providerconfig(providerConfig clusterv1.ProviderConfi
 	return &config, nil
 }
 
-func (oc *OpenstackClient) getKubeadmToken() (string, error) {
-	tokenParams := kubeadm.TokenCreateParams{
-		Ttl: time.Duration(10) * time.Minute,
+func (oc *OpenstackClient) createBootstrapToken() (string, error) {
+	token, err := tokenutil.GenerateBootstrapToken()
+	if err != nil {
+		return "", err
 	}
-	output, err := oc.kubeadm.TokenCreate(tokenParams)
+
+	expiration := time.Now().UTC().Add(TokenTTL)
+	tokenSecret, err := bootstrap.GenerateTokenSecret(token, expiration)
 	if err != nil {
-		glog.Errorf("unable to create token: %v", err)
+		// If this happens, there is a bug somewhere
 		return "", err
 	}
 
-	soutput := strings.Split(output, "\n")
-	return strings.TrimSpace(soutput[len(soutput)-1]), err
+	err = oc.client.Create(context.TODO(), tokenSecret)
+	if err != nil {
+		return "", err
+	}
+
+	if tokenSecret == nil {
+		return "", errors.New("unable to create token")
```

Comment:
When can this happen? It sounds like if there's a problem generating the token `oc.client.Create` would return an error, right?